Framework/DataHandling/inc/MantidDataHandling/LoadSampleShape.h

```diff
@@ -0,0 +1,71 @@
+#ifndef DATAHANDLING_LOAD_SHAPE_H_
+#define DATAHANDLING_LOAD_SHAPE_H_
+
+#include "MantidAPI/IFileLoader.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+#include "MantidKernel/V3D.h"
+
+namespace Mantid {
+namespace DataHandling {
+/**  Load Shape into an instrument of a workspace
+
+     The following file types are supported
+
+       STL file with suffix .stl
+
+
+@author Karl Palmen ISIS;
+@date 26/02/2018
+
+Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+ */
+
+class DLLExport LoadSampleShape
+    : public API::IFileLoader<Kernel::FileDescriptor> {
```

Comment:
Why does this inherit from IFileLoader?  why have you done that?

Reply:
I had originally hoped one could load an STL file using the load button in MantidPlot, but I no longer think it would work in practice so will remove this feature.

Reply:
This would not make sense as LoadSampleShape modifies an existing workspace, rather than creating a new one from a file.